ember-amount-input/src/components/amount-input.ts

```diff
@@ -0,0 +1,180 @@
+
+  /**
+   * Specifies the maximum value for the input field
+   */
+  max?: number;
+
+  /**
+   * Specifies the number of decimals to use for the amount value
+   */
+  numberOfDecimal?: number;
+
+  /**
+   * The placeholder displayed in the input
+   */
+  placeholder?: string;
+
+  /**
+   * Specifies if the input field should be read-only
+   */
+  readonly?: boolean;
+
+  /**
+   * Specifies the number intervals for the input field
+   */
+  step?: number;
+
+  /**
+   * The callback function triggered when the input value is updated
+   */
+  update: (value: string) => void;
+
+  /**
+   * The value of the input. It should be updated using the `update` argument
+   */
+  value: number | string;
+}
+
+export interface AmountInputSignature {
+  Element: HTMLDivElement;
+  Args: AmountInputArgs;
+}
+
+/**
+ * An amount/money input component.
+ *
+ * @public
+ * @class AmountInput
+ *
+ * @example
+ *
+ * ```hbs
+ *  <AmountInput @value={{this.value}} @update={{fn (mut this.value)}} />
+ * ```
+ */
+export default class AmountInput extends Component<AmountInputSignature> {
+  /**
+   * The currency displayed in the input.
+   * Defaults to `EUR`.
+   */
+  get currency(): string {
+    return this._argOrDefault('currency', 'EUR');
+  }
+
+  /**
+   * A custom id applied on the input.
+   * Defaults to `amount-input`.
+   */
+  get inputId(): string {
+    return this._argOrDefault('inputId', 'amount-input');
+  }
+
+  /**
+   * Specifies the number of decimals to use for the amount value.
+   * Can be >= 0.
+   * Defaults to 2.
+   */
+  get numberOfDecimal(): number {
+    return this._argOrDefault('numberOfDecimal', 2);
+  }
+
+  /**
+   * The placeholder displayed in the input.
+   * Defaults to `0.00`.
+   */
+  get placeholder(): string {
+    return this._argOrDefault('placeholder', '0.00');
+  }
+
+  /**
+   * Specifies the number intervals for the input field.
+   * Can be >= 0.
+   * Defaults to 0.01.
+   */
+  get step(): number {
+    return this._argOrDefault('step', 0.01);
+  }
+
+  @action
+  onKeyDown(event: KeyboardEvent): boolean {
+    if (event.keyCode === KEY_CODE_E) {
+      event.preventDefault();
+      return false;
+    } else if (
+      this.numberOfDecimal === 0 &&
+      [KEY_CODE_FULLSTOP, KEY_CODE_COMMA].includes(event.keyCode)
+    ) {
+      event.preventDefault();
+      return false;
+    }
+
+    return true;
+  }
+
+  @action
+  onInput(event: Event): boolean {
+    if (!(event.target instanceof HTMLInputElement)) return false;
+
+    const { value } = event.target;
+    this.args.update(value);
+
+    return true;
+  }
+
+  @action
+  onFocusOut(event: FocusEvent): boolean {
+    if (event.target instanceof HTMLInputElement) {
+      const { valueAsNumber, value } = event.target;
+      if (value) {
+        this.args.update(valueAsNumber.toFixed(this.numberOfDecimal));
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  _argOrDefault<K extends keyof AmountInputArgs, T, D extends T>(
```

Comment:
I think this is a `private` member, so we should explicitly mark it as `private`.

A part from that, personal opinion: I'd avoid using the `_` as prefix - I think it was more a JS thing to signal the member is private, in this case I don't think we need it. In TS I use it on variables (e.g.: `const _min`) that I want to expose through getters or setters (e.g.: `get min() {...}`).

Reply:
Interesting! How should I do that? A simple jsdocs comment with `@private`? Or maybe use the `#` prefix? However, for the prefix, it will require us to install a new babel dependency to interpret it I think

Reply:
This should do the job (https://www.typescriptlang.org/docs/handbook/2/classes.html#private):

```suggestion
  private argOrDefault<K extends keyof AmountInputArgs, T, D extends T>(
```

Reply:
Unfortunately, this is forbidden with our current configuration 😄  See [this](https://github.com/qonto/ember-amount-input/blob/master/ember-amount-input/.eslintrc.cjs#L81-L85). So we have two options:
- update our set of lint rules to allow using `private`
- use the `#` prefix instead (but it will require installing an extra Babel plugin)

Reply:
To me, the best option would be to update our configuration. For example `ember-lottie` uses the `private` keyword as well. The drawback is that we need to align all addons on this. But it's a quick fix, so I'm fine about it. wdyt?

Reply:
> Unfortunately, this is forbidden with our current configuration 😄 See [this](https://github.com/qonto/ember-amount-input/blob/master/ember-amount-input/.eslintrc.cjs#L81-L85).

Woah, I totally missed this 😢 

> To me, the best option would be to update our configuration. For example ember-lottie uses the private keyword as well. The drawback is that we need to align all addons on this. But it's a quick fix, so I'm fine about it. wdyt?

Agreed, let's do this! 🥳

Reply:
This one has been fixed - we are now using the `private` keyword for `argOrDefault`

Reply:
Wonderful!